### PR TITLE
Autogenerate HTML reports for Code Coverage

### PR DESCRIPTION
**Pull request by nigurr**
* Code coverage HTML will be auto generated for Jacoco/Cobertura
* No longer accept "Custom" reports to view in Azure DevOps Coverage
* Users can continue to publish additional reports which will be non-viewable

### CODE REVIEW

File: Tasks/PublishCodeCoverageResultsV1/task.json [hunk 1]
```diff
@@ -43,13 +43,31 @@
...
+        },
         {
             "name": "reportDirectory",
             "type": "string",
             "label": "Report directory",
             "defaultValue": "",
             "required": false,
-            "helpMarkDown": "Path of the code coverage HTML report directory. The report directory is published for later viewing as an artifact of the build. The value may contain minimatch patterns. For example: `$(System.DefaultWorkingDirectory)/MyApp/**/site/cobertura`"
+            "helpMarkDown": "Path of the code coverage HTML report directory. The report directory is published for later viewing as an artifact of the build. The value may contain minimatch patterns. For example: `$(System.DefaultWorkingDirectory)/MyApp/**/site/cobertura`",
+            "visibleRule": "false = true"
```
Comment (madhurig) [inline Tasks/PublishCodeCoverageResultsV1/task.json]: I haven't seen this visibleRule before. Doesn't this mean reportDirectory will never be visible? That would be a breaking change in the task.

File: Tasks/PublishCodeCoverageResultsV1/task.json [hunk 2]
```diff
@@ -43,13 +43,31 @@
...
+        {
+            "name": "autogenerateHtmlReport",
+            "type": "boolean",
+            "label": "Generate Html report",
+            "defaultValue": "false",
+            "required": false,
+            "helpMarkDown": "Generate code coverage html report."
+        },
+        {
+            "name": "sourceDirectory",
```
Comment (acesiddhu) [inline Tasks/PublishCodeCoverageResultsV1/task.json]: >sourceDirectory [](start = 21, length = 15)

check with Prachi

File: Tasks/PublishCodeCoverageResultsV1/publishcodecoverageresults.ts [hunk 3]
```diff
@@ -94,11 +112,65 @@ function resolvePathToSingleItem(workingDirectory:string, pathInput: string, isD
...
+        const tmpDir = tl.getVariable('Agent.TempDirectory');
+        return tmpDir;
+    } catch (err) {
+        tl.warning(tl.loc('UpgradeAgentMessage'));
+        return os.tmpdir();
+    }
+}
+
+async function generateHtmlReport(summaryFile: string, targetDir: string, sourceFolder: string, workingDir: string) {
+    const dotnetPath = tl.which('dotnet', false);
```
Comment (acesiddhu) [inline Tasks/PublishCodeCoverageResultsV1/publishcodecoverageresults.ts]: >dotnet [](start = 33, length = 6)

full dot net scenarios?

**Comment by HolgerJeromin** [comment]
Just for reference: This was reverted in #9947 and readded with #9962

**Pull request merged**